### ISSUE TITLE
elasticsearch_opendistro_destination keeps creating new destinations on subsequent runs

### ISSUE THREAD

**Issue by hanselke** [phillbaker/terraform-provider-elasticsearch#167]
i have a tf backend that works.

using resource elasticsearch_opendistro_destination, it keeps creating a new destination everytime i run it, instead of recongizing its already been created via the state file.

**Comment by phillbaker** [comment]
Hello, can you please include the following:
* redacted version of the terraform resource configuration
* terraform version
* provider version
* elasticsearch version (and opendistro version if relevant)
* terraform logs by setting [`TF_LOG=info`](https://www.terraform.io/docs/internals/debugging.html)

**Comment by IronCore864** [comment]
I'm having similar issues with both `elasticsearch_opendistro_destination` and `elasticsearch_opendistro_monitor`; everytime I apply with the same configuration it prompts changes.

```
Terraform v0.15.2
on darwin_amd64
+ provider registry.terraform.io/phillbaker/elasticsearch v1.5.5
```

Elasticsearch version7.10 (AWS managed ES, so opendistro I think)

Logs (INFO):
```
021-05-06T11:19:15.613+0200 [INFO]  Terraform version: 0.15.2
2021-05-06T11:19:15.613+0200 [INFO]  Go runtime version: go1.16.2
2021-05-06T11:19:15.613+0200 [INFO]  CLI args: []string{"/usr/local/bin/terraform", "apply"}
2021-05-06T11:19:15.614+0200 [INFO]  CLI command args: []string{"apply"}
2021-05-06T11:19:15.725+0200 [INFO]  Failed to read plugin lock file .terraform/plugins/darwin_amd64/lock.json: open .terraform/plugins/darwin_amd64/lock.json: no such file or directory
2021-05-06T11:19:15.725+0200 [INFO]  backend/local: starting Apply operation
2021-05-06T11:19:15.728+0200 [INFO]  provider: configuring client automatic mTLS
2021-05-06T11:19:15.778+0200 [INFO]  provider.terraform-provider-elasticsearch_v1.5.5: configuring server automatic mTLS: timestamp=2021-05-06T11:19:15.778+0200
2021-05-06T11:19:15.854+0200 [INFO]  terraform: building graph: GraphTypeValidate
2021-05-06T11:19:15.855+0200 [INFO]  provider: configuring client automatic mTLS
2021-05-06T11:19:15.904+0200 [INFO]  provider.terraform-provider-elasticsearch_v1.5.5: configuring server automatic mTLS: timestamp=2021-05-06T11:19:15.904+0200
2021-05-06T11:19:15.984+0200 [INFO]  backend/local: apply calling Plan
2021-05-06T11:19:15.984+0200 [INFO]  terraform: building graph: GraphTypePlan
2021-05-06T11:19:15.986+0200 [INFO]  provider: configuring client automatic mTLS
2021-05-06T11:19:16.033+0200 [INFO]  provider.terraform-provider-elasticsearch_v1.5.5: configuring server automatic mTLS: timestamp=2021-05-06T11:19:16.033+0200
2021-05-06T11:19:16.108+0200 [WARN]  ValidateProviderConfig from "provider[\"registry.terraform.io/phillbaker/elasticsearch\"]" changed the config value, but that value is unused
elasticsearch_opendistro_destination.opsgenie: Refreshing state... [id=XJ_0QHkBA2_qOif1BgtB]
2021-05-06T11:19:17.508+0200 [WARN]  Provider "registry.terraform.io/phillbaker/elasticsearch" produced an invalid plan for elasticsearch_opendistro_destination.opsgenie, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .body: planned value cty.StringVal("{\"custom_webhook\":{\"header_params\":{\"Authorization\":\"GenieKey xxx\",\"Content-Type\":\"application/json\"},\"url\":\"https://api.opsgenie.comv2/alerts\"},\"name\":\"opsgenie\",\"type\":\"custom_webhook\"}") does not match config value cty.StringVal("{\n  \"type\": \"custom_webhook\",\n  \"name\": \"opsgenie\",\n  \"custom_webhook\": {\n    \"url\": \"https://api.opsgenie.comv2/alerts\",\n    \"header_params\": {\n      \"Content-Type\": \"application/json\",\n      \"Authorization\": \"GenieKey xxx\"\n    }\n  }\n}\n") nor prior value cty.StringVal("{\"custom_webhook\":{\"header_params\":{\"Authorization\":\"GenieKey xxx\",\"Content-Type\":\"application/json\"},\"host\":null,\"method\":null,\"password\":null,\"path\":null,\"port\":-1,\"query_params\":{},\"scheme\":null,\"url\":\"https://api.opsgenie.comv2/alerts\",\"username\":null},\"id\":\"XJ_0QHkBA2_qOif1BgtB\",\"last_update_time\":1620292511857,\"name\":\"opsgenie\",\"primary_term\":0,\"schema_version\":3,\"seq_no\":0,\"type\":\"custom_webhook\",\"user\":{\"backend_roles\":[],\"custom_attribute_names\":[],\"name\":\"\",\"roles\":[],\"user_requested_tenant\":null}}")
elasticsearch_opendistro_monitor.test: Refreshing state... [id=X5_0QHkBA2_qOif16wsw]
2021-05-06T11:19:18.047+0200 [WARN]  Provider "registry.terraform.io/phillbaker/elasticsearch" produced an invalid plan for elasticsearch_opendistro_monitor.test, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .body: planned value cty.StringVal("{\"enabled\":true,\"inputs\":[{\"search\":{\"indices\":[\"logstash-*\"],\"query\":{\"aggregations\":{},\"query\":{\"bool\":{\"adjust_pure_negative\":true,\"boost\":1,\"filter\":[{\"range\":{\"@timestamp\":{\"boost\":1,\"format\":\"epoch_millis\",\"from\":\"{{period_end}}||-7d\",\"include_lower\":true,\"include_upper\":true,\"to\":\"{{period_end}}\"}}}]}},\"size\":0}}}],\"name\":\"OpsGenie-Test-Monitor\",\"schedule\":{\"period\":{\"interval\":1,\"unit\":\"MINUTES\"}},\"triggers\":[{\"actions\":[{\"destination_id\":\"XJ_0QHkBA2_qOif1BgtB\",\"message_template\":{\"lang\":\"mustache\",\"source\":\"{\\\"message\\\": \\\"Monitor {{ctx.monitor.name}} just entered alert status. Please investigate the issue. Trigger: {{ctx.trigger.name}} Severity: {{ctx.trigger.severity}} Period start: {{ctx.periodStart}} Period end: {{ctx.periodEnd}}\\\", \\\"alias\\\": \\\"Life is too short for no alias\\\"}\"},\"name\":\"OpsGenie-Test-Action\",\"subject_template\":{\"lang\":\"mustache\",\"source\":\"\"},\"throttle_enabled\":false}],\"condition\":{\"script\":{\"lang\":\"painless\",\"source\":\"ctx.results[0].hits.total.value \\u003e 1\"}},\"name\":\"OpsGenie-Test-Trigger\",\"severity\":\"1\"}],\"type\":\"monitor\"}") does not match config value cty.StringVal("{\n    \"name\": \"OpsGenie-Test-Monitor\",\n    \"type\": \"monitor\",\n    \"enabled\": true,\n    \"schedule\": {\n        \"period\": {\n            \"interval\": 1,\n            \"unit\": \"MINUTES\"\n        }\n    },\n    \"inputs\": [\n        {\n            \"search\": {\n                \"indices\": [\n                    \"logstash-*\"\n                ],\n                \"query\": {\n                    \"size\": 0,\n                    \"query\": {\n                        \"bool\": {\n                            \"filter\": [\n                                {\n                                    \"range\": {\n                                        \"@timestamp\": {\n                                            \"from\": \"{{period_end}}||-7d\",\n                                            \"to\": \"{{period_end}}\",\n                                            \"include_lower\": true,\n                                            \"include_upper\": true,\n                                            \"format\": \"epoch_millis\",\n                                            \"boost\": 1.0\n                                        }\n                                    }\n                                }\n                            ],\n                            \"adjust_pure_negative\": true,\n                            \"boost\": 1.0\n                        }\n                    },\n                    \"aggregations\": {}\n                }\n            }\n        }\n    ],\n    \"triggers\": [\n        {\n            \"name\": \"OpsGenie-Test-Trigger\",\n            \"severity\": \"1\",\n            \"condition\": {\n                \"script\": {\n                    \"source\": \"ctx.results[0].hits.total.value > 1\",\n                    \"lang\": \"painless\"\n                }\n            },\n            \"actions\": [\n                {\n                    \"name\": \"OpsGenie-Test-Action\",\n                    \"destination_id\": \"XJ_0QHkBA2_qOif1BgtB\",\n                    \"message_template\": {\n                        \"source\": \"{\\\"message\\\": \\\"Monitor {{ctx.monitor.name}} just entered alert status. Please investigate the issue. Trigger: {{ctx.trigger.name}} Severity: {{ctx.trigger.severity}} Period start: {{ctx.periodStart}} Period end: {{ctx.periodEnd}}\\\", \\\"alias\\\": \\\"Life is too short for no alias\\\"}\",\n                        \"lang\": \"mustache\"\n                    },\n                    \"throttle_enabled\": false,\n                    \"subject_template\": {\n                        \"source\": \"\",\n                        \"lang\": \"mustache\"\n                    }\n                }\n            ]\n        }\n    ]\n}\n") nor prior value cty.StringVal("{\"enabled\":true,\"inputs\":[{\"search\":{\"indices\":[\"logstash-*\"],\"query\":{\"aggregations\":{},\"query\":{\"bool\":{\"adjust_pure_negative\":true,\"boost\":1,\"filter\":[{\"range\":{\"@timestamp\":{\"boost\":1,\"format\":\"epoch_millis\",\"from\":\"{{period_end}}||-7d\",\"include_lower\":true,\"include_upper\":true,\"to\":\"{{period_end}}\"}}}]}},\"size\":0}}}],\"name\":\"OpsGenie-Test-Monitor\",\"schedule\":{\"period\":{\"interval\":1,\"unit\":\"MINUTES\"}},\"triggers\":[{\"actions\":[{\"destination_id\":\"XJ_0QHkBA2_qOif1BgtB\",\"id\":\"aZ_1QHkBA2_qOif1rAvF\",\"message_template\":{\"lang\":\"mustache\",\"source\":\"{\\\"message\\\": \\\"Monitor {{ctx.monitor.name}} just entered alert status. Please investigate the issue. Trigger: {{ctx.trigger.name}} Severity: {{ctx.trigger.severity}} Period start: {{ctx.periodStart}} Period end: {{ctx.periodEnd}}\\\", \\\"alias\\\": \\\"Life is too short for no alias\\\"}\"},\"name\":\"OpsGenie-Test-Action\",\"subject_template\":{\"lang\":\"mustache\",\"source\":\"\"},\"throttle_enabled\":false}],\"condition\":{\"script\":{\"lang\":\"painless\",\"source\":\"ctx.results[0].hits.total.value \\u003e 1\"}},\"id\":\"aJ_1QHkBA2_qOif1rAvF\",\"name\":\"OpsGenie-Test-Trigger\",\"severity\":\"1\"}],\"type\":\"monitor\",\"user\":{\"backend_roles\":[],\"custom_attribute_names\":[],\"name\":\"\",\"roles\":[],\"user_requested_tenant\":null}}")

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # elasticsearch_opendistro_destination.opsgenie will be updated in-place
  ~ resource "elasticsearch_opendistro_destination" "opsgenie" {
      ~ body = jsonencode(
          ~ {
              ~ custom_webhook   = {
                  - host          = null -> null
                  - method        = null -> null
                  - password      = null -> null
                  - path          = null -> null
                  - port          = -1 -> null
                  - query_params  = {} -> null
                  - scheme        = null -> null
                  - username      = null -> null
                    # (2 unchanged elements hidden)
                }
              - id               = "XJ_0QHkBA2_qOif1BgtB" -> null
              - last_update_time = 1620292511857 -> null
              - primary_term     = 0 -> null
              - schema_version   = 3 -> null
              - seq_no           = 0 -> null
              - user             = {
                  - backend_roles          = []
                  - custom_attribute_names = []
                  - name                   = ""
                  - roles                  = []
                  - user_requested_tenant  = null
                } -> null
                # (2 unchanged elements hidden)
            }
        )
        id   = "XJ_0QHkBA2_qOif1BgtB"
    }

  # elasticsearch_opendistro_monitor.test will be updated in-place
  ~ resource "elasticsearch_opendistro_monitor" "test" {
      ~ body = jsonencode(
          ~ {
              ~ triggers = [
                  ~ {
                      ~ actions   = [
                          ~ {
                              - id               = "aZ_1QHkBA2_qOif1rAvF" -> null
                                # (5 unchanged elements hidden)
                            },
                        ]
                      - id        = "aJ_1QHkBA2_qOif1rAvF" -> null
                        # (3 unchanged elements hidden)
                    },
                ]
              - user     = {
                  - backend_roles          = []
                  - custom_attribute_names = []
                  - name                   = ""
                  - roles                  = []
                  - user_requested_tenant  = null
                } -> null
                # (5 unchanged elements hidden)
            }
        )
        id   = "X5_0QHkBA2_qOif16wsw"
    }

Plan: 0 to add, 2 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value:
```

**Comment by phillbaker** [comment]
@IronCore864 a perpetual diff is a separate issue than what's originally described here, however, some fixes were merged to deal with diffs on the monitor resource in v1.5.6. Please try that version and open a new issue if it's still occurring.

I'm going to close this issue as a dupe of #158, which should be fixed in v1.5.6.
